### ISSUE TITLE
Help me get the data.

### ISSUE THREAD

**Issue by DmitrySkibitsky** [staudenmeir/eloquent-json-relations#38]
Hi everyone, I'm trying to add relationships, but it doesn't work.

models
```php
class JobVacancy extends Model implements Transformable
{
    use TransformableTrait;

    use HasJsonRelationships, Translatable {
        HasJsonRelationships::getAttribute as getAttributeJson;
        Translatable::getAttribute as getAttributeTranslations;
    }

    public function getAttribute($key)
    {
        if (! $this->isTranslationAttribute($key)) {
            return $this->getAttributeJson($key);
        }

        return $this->getAttributeTranslations($key);
    }

    protected $casts = [
        'parameters' => 'json',
    ];

    public $translatedAttributes = [
        'name',
        'description',
        'm_title',
        'm_description',
        'm_keywords',
    ];

    public function parameters()
    {
        return $this->belongsToJson('App\Entities\Parameter', 'parameters[]->id');
    }
}

class Parameter extends Model implements Transformable
{
    use TransformableTrait;
    use NodeTrait;

    use HasJsonRelationships, Translatable {
        HasJsonRelationships::getAttribute as getAttributeJson;
        Translatable::getAttribute as getAttributeTranslations;
    }

    public function getAttribute($key)
    {
        if (! $this->isTranslationAttribute($key)) {
            return $this->getAttributeJson($key);
        }

        return $this->getAttributeTranslations($key);
    }

    public $timestamps = false;

    public $translatedAttributes = [
        'name',
    ];

    public function __construct(array $attributes = [])
    {
        parent::__construct($attributes);

        $this->defaultLocale = app()->getLocale();
    }

    public function jobVacancies()
    {
        return $this->hasManyJson('App\Entities\JobVacancy', 'parameters[]->id');
    }
}
```

Tables
![Аннотация 2020-05-16 183125](https://user-images.githubusercontent.com/20288556/82125353-23dcd480-97ae-11ea-962e-724240fb046a.png)

Migrations
```php
        Schema::create(
            'job_vacancies',
            function (Blueprint $table) {
                $table->bigIncrements('id');
                $table->json('parameters')->nullable();
                $table->boolean('status')->default(true);
                $table->timestamps();
            }
        );

       Schema::create(
            'parameters',
            function (Blueprint $table) {
                $table->bigIncrements('id');
                $table->integer('sorting_order')->default(0);
                $table->nestedSet();
            }
        );
```

Data in DB
![Аннотация 2020-05-16 194617](https://user-images.githubusercontent.com/20288556/82125337-fe4fcb00-97ad-11ea-9e4f-258065245c94.png)
![Аннотация 2020-05-16 194643](https://user-images.githubusercontent.com/20288556/82125335-fc860780-97ad-11ea-9037-808286e36ad1.png)

Result
```log
App\Entities\JobVacancy {#1656 ▼
  #fillable: array:8 [▶]
  #casts: array:8 [▶]
  +translatedAttributes: array:5 [▶]
  #with: array:1 [▶]
  #connection: "mysql"
  #table: "job_vacancies"
  #primaryKey: "id"
  #keyType: "int"
  +incrementing: true
  #withCount: []
  #perPage: 15
  +exists: true
  +wasRecentlyCreated: false
  #attributes: array:12 [▶]
  #original: array:12 [▶]
  #changes: []
  #dates: []
  #dateFormat: null
  #appends: []
  #dispatchesEvents: []
  #observables: []
  #relations: array:2 [▼
    "images" => Illuminate\Database\Eloquent\Collection {#1658 ▶}
    "parameters" => Kalnoy\Nestedset\Collection {#1815 ▼
      #items: []
    }
  ]
  #touches: []
  +timestamps: true
  #hidden: []
  #visible: []
  #guarded: array:1 [▶]
  #defaultLocale: null
}
```

Please help us understand what the problem is

**Comment by staudenmeir** [comment]
What query are you executing to get this result? What are the `JobVacancy`'s attributes?

**Comment by DmitrySkibitsky** [comment]
> What query are you executing to get this result? What are the `JobVacancy`'s attributes?
```
`JobVacancy`'s attributes: 
protected $fillable = [
        'price',
        'price_end',
        'shortcuts',
        'parameter_heading_id',
        'parameter_location_id',
        'parameters',
        'status',
        'archive',
    ];

    protected $casts = [
        'price' => 'float',
        'price_end' => 'float',
        'shortcuts' => 'array',
        'parameter_heading_id' => 'integer',
        'parameter_location_id' => 'integer',
        'archive' => 'integer',
        'parameters' => 'json',
        'status' => 'integer',
    ];
```
query method
![Аннотация 2020-05-16 200526](https://user-images.githubusercontent.com/20288556/82125716-977fe100-97b0-11ea-818e-0df2c9358c7b.png)

dump
![Аннотация 2020-05-16 194643](https://user-images.githubusercontent.com/20288556/82125718-9b136800-97b0-11ea-93c8-d9dfd65fcb95.png)

**Comment by staudenmeir** [comment]
What's the result of `dd($vacancies->first()->getAttributes());`?

**Comment by DmitrySkibitsky** [comment]
> What's the result of `dd($vacancies->first()->getAttributes());`?

```log
array:12 [▼
  "id" => 1
  "price" => 50.0
  "price_end" => 50.0
  "shortcuts" => "[{"value":1}]"
  "parameter_heading_id" => 3
  "parameter_location_id" => 17
  "parameters" => "[{"id":17},{"id":22},{"id":13},{"id":9},{"id":6}]"
  "status" => 1
  "created_at" => "2020-05-08 11:46:00"
  "updated_at" => "2020-05-15 22:07:42"
  "sorting_order" => 999
  "archive" => 0
]
```

**Comment by staudenmeir** [comment]
Thanks, I'll look into it.

**Comment by DmitrySkibitsky** [comment]
> What's the result of `dd($vacancies->first()->getAttributes());`?

```
Laravel Framework 6.18.14

  "require": {
    "php": "^7.2",
    "ext-gd": "*",
    "ext-json": "*",
    "artesaos/seotools": "^0.17.2",
    "astrotomic/laravel-translatable": "^11.7",
    "fideloper/proxy": "^4.0",
    "fruitcake/laravel-cors": "2.0.x-dev",
    "itsgoingd/clockwork": "^4.0",
    "jenssegers/agent": "^2.6",
    "kalnoy/nestedset": "^5.0",
    "laravel/framework": "^6.2",
    "laravel/tinker": "^2.0",
    "mcamara/laravel-localization": "^1.5",
    "pherum/laravel-bbcode": "^1.0",
    "prettus/l5-repository": "^2.6",
    "staudenmeir/eloquent-json-relations": "^1.1",
    "tymon/jwt-auth": "^1.0.0-rc.5"
  },
```
The Parameter class has trait NodeTrait, but it is also impossible to get data without it. The only thing that changes after I remove this trait

before
```
"parameters" => Kalnoy\Nestedset\Collection {#1815 ▼
      #items: []
    }
```
after
```
 "parameters" => Illuminate\Database\Eloquent\Collection {#1809 ▼
      #items: []
    }
```

**Comment by DmitrySkibitsky** [comment]
> Thanks, I'll look into it.

Okay, thanks.

**Comment by staudenmeir** [comment]
I haven't been able to reproduce the issue so far.

Can you log the executed queries with `\DB::enableQueryLog(); [...] dd(\DB::getQueryLog());` or a package like https://github.com/barryvdh/laravel-debugbar?

**Comment by staudenmeir** [comment]
There is an issue you should fix: Both attribute and relationship are named `parameters`. This is problematic because it prevents you from accessing the relationship with `$vacancy->parameters`.

**Comment by DmitrySkibitsky** [comment]
> There is an issue you should fix: Both attribute and relationship are named `parameters`. This is problematic because it prevents you from accessing the relationship with `$vacancy->parameters`.

Aww... I hadn't thought of that. My apologies. I'll check later, because I probably won't be working today.

**Comment by DmitrySkibitsky** [comment]
> There is an issue you should fix: Both attribute and relationship are named `parameters`. This is problematic because it prevents you from accessing the relationship with `$vacancy->parameters`.

Good time of day. I found a problem that caused the package to not work. Despite the fact that the data was correct after the selection in the collection the model had mutations that I removed and everything started working. I'm sorry to have disturbed you.
![Аннотация 2020-05-17 105646444](https://user-images.githubusercontent.com/20288556/82139077-149d6b80-982e-11ea-9a73-b2b8899e02c0.png)
But the name will need to be changed in any case, as it is misleading.
